src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2463,12 +2465,49 @@ InitializeDistTableCache(void)
 
 
 /*
- * GetWorkerNodeHash is a wrapper around InitializeWorkerNodeCache(). It
- * triggers InitializeWorkerNodeCache when the workerHash is invalid. Otherwise,
- * it returns the hash.
+ * GetWorkerNodeArray returns the worker node data as an array of pointers to the
+ * WorkerNode structs allocated in WorkerNode hash. It also sets the integer pointed
+ * to by workerNodeCountResult to the current worker node count.
+ *
+ * The array is returned from the cache, if the cache is not (yet) valid, it is first
+ * rebuilt.
+ */
+static WorkerNode **
+GetWorkerNodeArray(int *workerNodeCountResult)
```

Comment:
This feels like unnecessary ceremony. Since this function is only called once, what about just inlining the `PrepareWorkerNodeCache()` call into `LookupNodeForGroup` and using `WorkerNode{Count,Array}` directly?